<launch>
  <arg name="base_link" default="soma_link"/>
  <arg name="camera" default="camera_L"/>

  <!-- degree of pan angle-->
  <arg name="pan" default="0"/>
  <!-- degree of tile angle-->
  <arg name="tilt" default="15"/>

  <include file="$(find realsense2_camera)/launch/rs_rgbd.launch">

    <arg name="camera" value="$(arg camera)"/>
    <arg name="external_manager" value="false"/>
    <arg name="manager" value="realsense2_manager_$(arg camera)"/>

    <!-- <arg name="serial_no" value="939622073512" /> -->
    <arg name="serial_no" value="843112071843" />

    <arg name="enable_fisheye" value="false"/>
    <arg name="enable_infra1" value="false"/>
    <arg name="enable_infra2" value="false"/>
    <arg name="enable_gyro" value="false"/>
    <arg name="enable_accel" value="false"/>
    <arg name="enable_pointcloud" value="false" />
    <arg name="color_fps" value="15"/>
    <arg name="depth_fps" value="15"/>
  </include>

  <group ns="camera_L">
    <!-- transform publisher -->
    <arg name="Pi" default="3.1415" />
    <arg name="Pi_Pan" default="$(eval arg('pan')/180.0*arg('Pi'))"/>
    <arg name="Pi_Tilt" default="$(eval  arg('tilt')/180.0*arg('Pi'))"/>
    <arg name="tf_param1" default="0 0.25 0 $(arg Pi_Pan) $(arg Pi_Tilt) 0"/>
    <arg name="tf_param2" default="0.85 0.0 0.8 0 0 0"/>

    <node pkg="tf2_ros" type="static_transform_publisher" name="Camera2Center" args="$(arg tf_param1) camera_center_link $(arg camera)_link"/>
    <node pkg="tf2_ros" type="static_transform_publisher" name="Center2Base" args="$(arg tf_param2) $(arg base_link) camera_center_link"/>

    <node pkg="nodelet" type="nodelet" name="prefiltering_nodelet" args="load soma_vision/PrefilteringNodelet /$(arg camera)/realsense2_manager_$(arg camera)">
      <remap from="input_points" to="/$(arg camera)/depth_registered/points"/>
      <param name="use_distance_filter" value="true"/>
      <param name="distance_near_thresh" value="0.1"/>
      <param name="distance_far_thresh" value="6.0"/>
      <param name="down_sample_leaf_size" value="0.03"/>
      <param name="sor_k" value="10"/>
      <param name="sor_stddev" value="0.5"/>
    </node>

  </group>
  
</launch>
